packages/clerk-js/src/core/resources/BackupCode.ts

```diff
@@ -16,11 +16,13 @@ export class BackupCode extends BaseResource implements BackupCodeResource {
     this.fromJSON(data);
   }
 
-  protected fromJSON(data: BackupCodeJSON): this {
-    this.id = data.id;
-    this.codes = data.codes;
-    this.updatedAt = unixEpochToDate(data.updated_at);
-    this.createdAt = unixEpochToDate(data.created_at);
+  protected fromJSON(data: BackupCodeJSON | null): this {
+    if (data) {
+      this.id = data.id;
+      this.codes = data.codes;
+      this.updatedAt = unixEpochToDate(data.updated_at);
+      this.createdAt = unixEpochToDate(data.created_at);
+    }
```

Comment:
ℹ️ Refactored all the resources to handle `null` instead of a data object in their `fromJSON`

Reply:
❓ why? Do we call the `fromJSON()` when there is no data?

Reply:
@dimkl This is a valid question. The default mechanism when calling requests in `BaseResource` (and in extend to all the other resources) is to call `fromJSON()` after getting the response to update the resource data. 

Currently, if the `fetch` throws an error, the execution thread stops, and the error reaches the browser.

After this PR, if `fetch` fails, in some cases we mute the error and return `null`. Maybe we want to catch it earlier. Open to suggestions. :)

Reply:
So `null` means that no change will be made to the state of the app?